atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/KvTableMappingService.java

```diff
@@ -110,14 +115,31 @@ private TableReference getAlreadyExistingMappedTableName(TableReference tableRef
 
     @Override
     public void removeTable(TableReference tableRef) {
-        Cell key = Cell.create(getBytesForTableRef(tableRef), AtlasDbConstants.NAMESPACE_SHORT_COLUMN_BYTES);
+        removeTables(ImmutableSet.of(tableRef));
+    }
+
+    @Override
+    public void removeTables(Set<TableReference> tableRefs) {
         if (kv.getAllTableNames().contains(AtlasDbConstants.NAMESPACE_TABLE)) {
-            kv.delete(AtlasDbConstants.NAMESPACE_TABLE, ImmutableMultimap.of(key, 0L));
+            Multimap<Cell, Long> deletionMap = tableRefs.stream()
+                    .map(tableRef -> Cell.create(getBytesForTableRef(tableRef),
+                            AtlasDbConstants.NAMESPACE_SHORT_COLUMN_BYTES))
+                    .collect(Multimaps.toMultimap(
+                            key -> key,
+                            unused -> AtlasDbConstants.TRANSACTION_TS,
+                            MultimapBuilder.hashKeys().arrayListValues()::build));
+            kv.delete(AtlasDbConstants.NAMESPACE_TABLE, deletionMap);
+        }
+
+        // Need to invalidate the table refs in case we end up re-creating the same table again.
+        while (true) {
+            BiMap<TableReference, TableReference> existingMap = tableMap.get();
+            BiMap<TableReference, TableReference> newMap = HashBiMap.create(existingMap);
+            tableRefs.forEach(newMap::remove);
```

Comment:
This will drop updates from other threads no? We'd eventually get there I *think*.

`MapDifference` usually helps, but not sure how to deal with table removals without some sort of sentinel, and also don't want to keep them in the map long 🤷‍♂️

Reply:
It will drop updates from other threads, and if I understand things correctly, we may even fail to eventually get the right answer. 

Basically with this PR I gave up any guarantees of consistency if you concurrently drop and create the same table, the main thing I wanted to do was to ensure that concurrent updates to table B don't affect table A.